### PR TITLE
fix: Add Spark-compatible split function, as our current one does not handle regex

### PR DESCRIPTION
**Pull request by EmilyFlarionIO**
Adds a regex-compatible split function that mirrors Apache Spark's behavior when splitting strings with regex patterns. The current implementation only handles literal string delimiters.

### CODE REVIEW

File: crates/polars-plan/src/dsl/string.rs [hunk 1]
```diff
@@ -637,4 +638,10 @@ impl StringNameSpace {
             Some(Default::default()),
         )
     }
+
+    /// Split the string by a substring regex. The resulting dtype is `List<String>`.
+    pub fn flarion_split(self, by: Expr, n: Expr) -> Expr {
```
Comment (shanyfl) [inline crates/polars-plan/src/dsl/string.rs]: Maybe rename to `flarion_split_by_regex` for more clarity?

**Review by ranflarion — APPROVED** [review]
good stuff

File: crates/polars-plan/src/dsl/string.rs (hunk 1 above)
Comment (EmilyFlarionIO) [reply crates/polars-plan/src/dsl/string.rs]: The idea is to recreate the split() function from Spark, whenever we create a Spark function in polars we use the name flarion_<spark_function_name>, like in flarion_slice

Comment (shanyfl) [reply crates/polars-plan/src/dsl/string.rs]: Got it 👌

**Review by shanyfl — APPROVED** [review]

**Comment by EmilyFlarionIO** [comment]
Apparently I'm over engineering this, will revisit momentarily

**Comment by EmilyFlarionIO** [comment]
dynamic flarion_split (which allows columns as input) is saved as branch flarion-0.43-dynamic_split
In case we ever need to revisit this, or if we want to submit a PR to polars or smth, I think it was a good polars function ngl

File: crates/polars-ops/src/chunked_array/strings/flarion_split.rs [hunk 2]
```diff
@@ -0,0 +1,181 @@
+use arrow::array::ValueSize;
+use regex::RegexBuilder;
+
+use super::*;
+
+// I'm hoping the compiler optimizes this out, but it is required for correct outputs
+#[inline]
```
Comment (shanyfl) [inline crates/polars-ops/src/chunked_array/strings/flarion_split.rs:7]: #[inline(always)] ? 😭

File: crates/polars-ops/src/chunked_array/strings/flarion_split.rs [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            ca.into_iter().fold(&mut builder, |builder, opt_s| {
+                match opt_s {
+                    Some(s) => builder.append_values_iter(fill_empty_iter(split_chars(s))),
+                    None => builder.append_null(),
+                };
+                builder
+            });
+        };
+    } else {
+        let re = RegexBuilder::new(pattern).size_limit(31457280).build()?;
```
Comment (shanyfl) [inline crates/polars-ops/src/chunked_array/strings/flarion_split.rs]: I think this is more readable:

```rust
        let re = RegexBuilder::new(pattern).size_limit(30*1024*1024).build()?;
```

File: crates/polars-ops/src/chunked_array/strings/flarion_split.rs (hunk 2 above)
Comment (EmilyFlarionIO) [reply crates/polars-ops/src/chunked_array/strings/flarion_split.rs:7]: LLVM pretty much straight up ignores inline(always), I was actually more referring to the collect() and then into_iter() calls^^

File: crates/polars-ops/src/chunked_array/strings/flarion_split.rs (hunk 3 above)
Comment (EmilyFlarionIO) [reply crates/polars-ops/src/chunked_array/strings/flarion_split.rs]: 100%

**Review by shanyfl — APPROVED** [review]
Great work 👏

**Pull request merged**